tensorflow/python/eager/function.cc

```diff
@@ -0,0 +1,84 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <Python.h>
+
+#include "pybind11/pybind11.h"
+#include "pybind11/stl_bind.h"
+
+struct ConcreteFunction;  // Forward declaration.
+
+// TODO(jlchu): Migrate Python characteristics to C++
+
+namespace tensorflow {
+
+namespace py = pybind11;
+
+py::module* np_arrays = new py::module(
+    py::module::import("tensorflow.python.ops.numpy_ops.np_arrays"));
+py::module* nest = new py::module(
+    py::module::import("tensorflow.python.util.nest"));
+
+struct ConcreteFunction {
+  ConcreteFunction() {};
+  py::object _build_call_outputs(py::object result,
+                                 py::object structured_outputs,
+                                 bool _ndarrays_list,
+                                 bool _ndarray_singleton);
+};
+
+py::object ConcreteFunction::_build_call_outputs(py::object result,
+                                                 py::object structured_outputs,
+                                                 bool _ndarrays_list,
+                                                 bool _ndarray_singleton) {
+  if (structured_outputs.is_none()) {
+    return result;
+  }
+
+  // TODO(jlchu): Verify invariant -result = None only if
```

Comment:
what is the plan for this TODO? can we resolve in this change

Reply:
On my part, basically this would have amounted to looking at the preceding functions in the call stack more closely to verify that this sort of case is indeed covered. From the previous Python code (and a test case I came across), this seems to be implied.  I planned to leave the code as is for now, as long as tests are passing, as I'm not sure how long it would take to go back to verify it symbolically. Maybe I can add an assert statement for the time being?

I put the comment at this point in the code because a `py::none` cannot be casted to a list. The logic is the same as the Python but I guess the C++ exposes this a little more.

Thoughts?